addon_service/common/hmac.py

```diff
@@ -67,37 +54,145 @@ def make_signed_headers(
     )
 
 
-def _reconstruct_string_to_sign_from_request(request, signed_headers: list[str]) -> str:
-    signed_segments = [request.method, request.path]
-    query_string = request.META.get("QUERY_STRING")
-    if query_string:
-        signed_segments.append(query_string)
-    signed_segments.extend(
-        [str(request.headers[signed_header]) for signed_header in signed_headers]
-    )
-    return "\n".join([segment for segment in signed_segments if segment])
-
-
-def validate_signed_headers(request, hmac_key=None):
-    match = _AUTH_HEADER_REGEX.match(request.headers.get("Authorization", ""))
+@functools.lru_cache
+def validate_signed_headers(
+    request: HttpRequest, hmac_key: str, expiration_seconds: int | None = None
+) -> dict[str, str]:
+    _authorization = request.headers.get("Authorization", "")
+    if not _authorization.startswith(_AUTH_HEADER_PREFIX):
```

Comment:
Are all HMAC-authenticated requests guaranteed to have an `Authorization` field in the header that starts with `HMAC-`? I have been looking online and there does not seem to be a standard. In which case it is an agreement between the sender (OSF) and the receiver (GV). If that is the case, does it make sense to have `_AUTH_HEADER_PREFIX` to be something imported from local env vars shared by both OSF and GV?

Reply:
oh good point, i assumed wrong about the `HMAC-` prefix without checking for a standard -- correct we need only agreement with OSF (another reason shared code across services would be helpful), will update the errors accordingly (and perhaps add environment variables, yeah)